File: Leetcode/sbgo/Leetcode375-猜数字大小.go
package sbgo

func getMoneyAmount(n int) int {
	f := make([][]int, n+2)
	for i := range f {
		f[i] = make([]int, n+2)
	}
	for len := 2; len <= n; len++ {
		for l := 1; l+len-1 <= n; l++ {
			r := l + len - 1
			f[l][r] = int(1e9)
			for k := l; k <= r; k++ {
				f[l][r] = min(f[l][r], max(f[l][k-1], f[k+1][r])+k)
			}
		}
	}
	return f[1][n]
}

File: Leetcode/PY/Leetcode375-猜数字大小.py
class Solution:
    def getMoneyAmount(self, n: int) -> int:
        f = [[0] * (n + 2) for _ in range(n+2)]
        for len in range(2,n + 1):
            l = 1
            while l + len - 1 <= n:
                r = l + len - 1
                f[l][r] = int(1e9)
                for k in range(l,r + 1):
                    f[l][r] = min(f[l][r],max(f[l][k - 1],f[k + 1][r]) + k)
                l += 1
        return f[1][n]
